dace/frontend/fortran/ast_transforms.py

```diff
@@ -1,38 +1,113 @@
-# Copyright 2023 ETH Zurich and the DaCe authors. All rights reserved.
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
 
-from dace.frontend.fortran import ast_components, ast_internal_classes
-from typing import Dict, List, Optional, Tuple, Set
 import copy
+import re
+import warnings
+from collections import namedtuple
+from typing import Dict, List, Optional, Tuple, Set, Union, Type
 
+import sympy as sp
 
-def iter_fields(node: ast_internal_classes.FNode):
-    """
-    Yield a tuple of ``(fieldname, value)`` for each field in ``node._fields``
-    that is present on *node*.
-    """
-    if not hasattr(node, "_fields"):
-        a = 1
-    for field in node._fields:
-        try:
-            yield field, getattr(node, field)
-        except AttributeError:
-            pass
+from dace import symbolic as sym
+from dace.frontend.fortran import ast_internal_classes, ast_utils
+from dace.frontend.fortran.ast_desugaring import ConstTypeInjection
+from dace.frontend.fortran.ast_internal_classes import Var_Decl_Node, Name_Node, Int_Literal_Node, Data_Ref_Node, \
+    Execution_Part_Node, Array_Subscript_Node, Bool_Literal_Node
+from dace.frontend.fortran.ast_utils import mywalk, iter_fields, iter_attributes, TempName, singular, atmost_one, \
+    match_callsite_args_to_function_args
 
 
-def iter_child_nodes(node: ast_internal_classes.FNode):
-    """
-    Yield all direct child nodes of *node*, that is, all fields that are nodes
-    and all items of fields that are lists of nodes.
-    """
+class NeedsTypeInferenceException(BaseException):
+
+    def __init__(self, func_name, line_number):
+        self.line_number = line_number
+        self.func_name = func_name
+
+
+class Structure:
```

Comment:
what's this? if it's a data structure like dace.data.Structure, please define properly and even use a more descriptive name (ASTStructure? or something). Additionally, if it is what I think it is, vars should NOT be a dictionary, but an ordered list of tuples (because order matters!)